File: mobile/iosApp/iosApp/Screens/Content/ContentLoadFailureScreen.swift
//
//  ContentLoadFailureScreen.swift
//  iosApp
//
//  Created by Szymon Klimek on 30.10.24.
//  Copyright © 2024 orgName. All rights reserved.
//

import app
import SwiftUI

struct ContentLoadFailureScreen: View {
    let state: ManageContentState.Failure
    let actionSender: ActionSender
    
    init(state: ManageContentState.Failure, actionSender: ActionSender) {
        self.state = state
        self.actionSender = actionSender
    }
    
    var body: some View {
        VStack {
            Text("Manage content failure screen")
            Button("Retry download content") {
                actionSender.sendAction(action: ManageContentAction.DownloadUnitsTapped())
            }
        }
    }
}

File: mobile/androidApp/src/main/kotlin/com/langsapp/android/ui/content/ContentLoadFailureScreen.kt
package com.langsapp.android.ui.content

import androidx.compose.foundation.layout.Column
import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import com.langsapp.architecture.Action
import com.langsapp.architecture.ActionSender
import com.langsapp.content.ManageContentAction
import com.langsapp.content.ManageContentState

@Composable
fun ContentLoadFailureScreen(
    actionSender: ActionSender<Action>,
    state: ManageContentState.Failure,
) {
    Column {
        Text("Manage content failure screen")
        Button(
            onClick = { actionSender.sendAction(ManageContentAction.DownloadUnitsTapped) },
        ) {
            Text("Retry download content")
        }
    }
}
